File: 2015/01/aoc2015_01.py
#!/usr/bin/env python
import sys

def chars(io):
    while True:
        c = io.read(1)
        if not c: break
        yield c

def delta(c): return {'(': 1, ')': -1}.get(c, 0)

accum = 0
negind = None
for i, d in enumerate(map(delta, chars(sys.stdin))):
    accum = accum + d
    if accum < 0 and negind is None: negind = i + 1
print(accum, negind)

File: 2015/01/aoc2015_01.js
#!/usr/bin/env node
var accum = 0, i = 0, negind = null;
process.stdin.on('readable', () => {
	while((c = process.stdin.read(1)) !== null) {
		if(c == '(') accum += 1;
		else if (c == ')') accum -= 1;
		if(++i && (accum < 0) && (negind === null)) negind = i;
	}
});
process.stdin.on('end', () => { process.stdout.write(`${accum} ${negind}\n`); });
